### PR TITLE
Better input validation for nsolve #12052

### PR DESCRIPTION
**Pull request by tanayag**
<!-- If this pull request fixes an issue please indicate which issue by typing "Fixes #NNNN" below.>

#12052

### CODE REVIEW

**Comment by asmeurer** [comment]
You are assuming f is an iterable but it can be an expression.

File: sympy/solvers/solvers.py [hunk 1]
```diff
@@ -2750,6 +2750,9 @@ def nsolve(*args, **kwargs):
         f = args[0]
         fargs = None
         x0 = args[1]
+        for func0 in f:
+            if len(func0.free_symbols)>1:
```
Comment (asmeurer) [inline sympy/solvers/solvers.py]: This test isn't correct. If f is an iterable and only two arguments are given, the function should error, regardless of how many free_symbols there are (three arguments need to be given in this case).

**Comment by asmeurer** [comment]
You can use `raises` to add a test for the change.

File: sympy/solvers/solvers.py (hunk 1 above)
Comment (tanayag) [inline sympy/solvers/solvers.py]: so should I remove whole elif test case
```
elif len(args) == 2:
        f = args[0]
        fargs = None
        x0 = args[1]

```
is that what u r saying?

Comment (tanayag) [inline sympy/solvers/solvers.py]: @asmeurer ??

**Pull request closed without merging**